### PR TITLE
6864 unique sync index

### LINKED ISSUE #6864: Generate unique key for remote sync
# Description

During explorations it was found that it is necessary to send a unique key, this is difficult when the tables have composite pks, for this the solution will be to calculate the hash of a text concatenated by all composite keys.

image

**Actual result**: Infinite remote sync.
**Expected result**: Correct remote sync.

# DoD
- [x] Modify create stmt in syscollector module
- [x] Inject during the insertion of a new record the hash of all pks for the table to be updated locally.
- [x] Modify manager queries, make the range filter with this field to add.
- [x] Modify Update/install .sql stmt for the new field in manager tables.
- [x] Modify fixed metadata of manager syscollector decoder/wazuhdb to insert new field.
- [x] Run RTR
- [x] PR with CI OK.

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
|#6864||
Closes #6864 
## **Description**

This issue aims to provide a unique key to identify items in tables that use compounded primary keys to be usable by rsync algorithm.

These changes have to be made in the various implementations according to the operating system targeted for this feature.

## **DoD**
- [x] Implementation
- [x] **Run syscollctor testtool on Win**
image
- [x] **Check syncronization on Win agent**
image
- [x] **Run syscollctor testtool on Linux**
image
- [x] **Check syncronization on Linux agent**
image
- [x] **RTR syscollector**
image
- [x] **RTR Utils**
image

### CODE REVIEW

File: src/shared_modules/utils/stringHelper.h [hunk 1]
```diff
@@ -103,7 +103,19 @@ namespace Utils
...
+        {
+            const auto size{asciiData.size() * 2};
+            const auto buffer{std::make_unique<char[]>(size + 1)};
+            char* output{buffer.get()};
+            for (const auto& value : asciiData)
+            {
+                snprintf(output, 3, "%02x", value);
+                output += 2;
+            }
+            return {buffer.get(), size};
```
Comment (Dwordcito) [inline src/shared_modules/utils/stringHelper.h:118]: single return per function.

File: src/wazuh_modules/wm_syscollector.c [hunk 2]
```diff
@@ -9,6 +9,8 @@
  * Foundation.
  */
 #include <stdlib.h>
+#include <openssl/ssl.h>
```
Comment (Dwordcito) [inline src/wazuh_modules/wm_syscollector.c:12]: remove openssl include files

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/syscollector/src/syscollectorImp.cpp [hunk 3]
```diff
@@ -660,8 +668,32 @@ constexpr auto NETADDR_SQL_STATEMENT
        netmask TEXT,
        broadcast TEXT,
        checksum TEXT,
+       item_id TEXT,
        PRIMARY KEY (iface,proto,address)) WITHOUT ROWID;)"
 };
+static const std::vector<std::string> NETADDRESS_ITEM_ID_FIELDS{"iface", "proto", "address"};
+
+
+static std::string getItemId(const nlohmann::json& item, const std::vector<std::string> idFields)
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/src/syscollectorImp.cpp:677]: `const &` for idFields?

**Review by Dwordcito — APPROVED** [review]

**Review by tinachoB — APPROVED** [review]
LGTM!

**Pull request merged**